packages/calcite-components/src/components/list/list.tsx

```diff
@@ -35,6 +35,12 @@ import {
 } from "../../utils/sortableComponent";
 import { SLOTS as STACK_SLOTS } from "../stack/resources";
 
+type ListDragDetail = DragDetail & {
```

Comment:
WDYT about enhancing `DragDetail` to use generics to define the to, from, drag element types? Something like:

```ts
export interface DragDetail<
  To extends HTMLElement = HTMLElement, 
  From extends HTMLElement = HTMLElement, 
  Drag extends HTMLElement = HTMLElement
> {
  toEl: To;
  fromEl: From;
  dragEl: Drag;
  newIndex: number;
  oldIndex: number;
}
```

Reply:
@jcfranco I'm seeing this error in the build:

```console
Error: angular-workspace:build: projects/component-library/src/lib/stencil-generated/components.ts:1410:52 - error TS2315: Type 'DragDetail' is not generic.
```

Reply:
It looks like a Stencil 2.x issue (see https://github.com/ionic-team/stencil-ds-output-targets/issues/314). We could test w/ v4 to see if the issue is fixed. If v4 doesn't look any different, we could proceed with your previous approach and revisit the interface once generics are supported.

Reply:
> proceed with your previous approach and revisit the interface once generics are supported.

This sounds like a good approach. We can try updating it when 4x is installed.